content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2416 @@
+previously appended histograms. This requires a complete re-encoding of the
+entire chunk. (There is some optimization potential in only re-encoding the
+affected parts. Implementing this would be quite complicated. So far, the
+performance impact of the full re-encoding did not stick out as problematic.)
+
+### Staleness markers
+
+NOTE: To understand the following section, it is important to recall how
+staleness markers work in the TSDB. Staleness markers in float series are
+represented by one specific bit pattern among the many that can be used to
+represent the `NaN` value. This very specific float value is called “special
+stale `NaN` value” in the following section. It is (almost certainly) never
+returned by the usual arithmetic float operations and as such different from a
+“naturally occurring” `NaN` value, including those discussed in [Special cases
+of observed values](#special-cases-of-observed-values). In fact, the special
+stale `NaN` value is never returned directly when querying the TSDB, but it is
+handled internally before it reaches the caller.
+
+To mark staleness in histogram series, the usual special stale `NaN` value
+could be used. However, this would require cutting a new chunk, just for the
+purpose of marking the series as stale, because a float value following a
+histogram value has to be stored in a different chunk (see above). Therefore,
+there is also a histogram version of a stale marker where the field for the sum
+of observations is set to the special stale `NaN` value. In this case, all
+other fields are ignored, which enables setting them to values suitable for
+efficient storage (as the histogram version of a stale marker is essentially
+just a storage optimization). This works for both float and integer histograms
+(as the sum field is a float value even in an integer histogram), and the
+appropriate version can be used to avoid cutting a new chunk. All version of a
+stale marker (float, integer histogram, float histogram) MUST be treated as
+equivalent by the TSDB.
+
+### Chunk size limit
+
+The size of float chunks is limited to 1024 bytes. The same size limitation is
+generally used for histogram chunks, too. However, individual histograms can
+become very large if they have many buckets, so blindly enforcing the size
+limit could lead to chunks with very few histograms. (In the most extreme case,
+a single histogram could even take more than 1024 bytes so that the size limit
+could not be enforced at all.) With very few histograms per chunk, the
+compression ratio becomes worse. Therefore, a minimum number of 10 histograms
+per chunks has to be reached before the size limit of 1024 bytes kicks in. This
+implies that histogram chunks can be much larger than 1024 bytes.
+
+Requiring a minimum of 10 histograms per chunk is an initial, very simplistic
+approach, which might be improved in the future to find a better trade-off
+between chunk size and compression ratio.
+
+### Counter reset considerations
```

Comment:
This is quite complex. What do you think about switching to Created Timestamp feature for it? Would it simplify it? We will have it for all counters soon.

Reply:
CT is in itself a new and experimental feature. I wouldn't conflate them.

Even if CT were already established, I don't think it can be relied upon for NH ingestion. We need to detect counter reset as part of the encoding anyway. CT is more an additional source of information: We can mark (and store) a counter reset even if it is not detectable by conventional means, see below.